filters/auth/jwt_validation.go

```diff
@@ -0,0 +1,301 @@
+package auth
+
+import (
+	"crypto/rsa"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"net/http"
+	"net/url"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-co-op/gocron"
+	"github.com/golang-jwt/jwt"
+	"github.com/opentracing/opentracing-go"
+	log "github.com/sirupsen/logrus"
+	"github.com/zalando/skipper/filters"
+)
+
+const (
+	JwtValidationName = "jwtValidation"
+)
+
+type (
+	jwtValidationSpec struct {
+		options TokenintrospectionOptions
+	}
+
+	jwtValidationFilter struct {
+		scheduler       *gocron.Scheduler
+		claims          []string
+		upstreamHeaders map[string]string
+	}
+)
+
+var m sync.Mutex
+var rsakeys map[string]*rsa.PublicKey = make(map[string]*rsa.PublicKey)
```

Comment:
I think for the filter to be usable we should support variety of algorithms (e.g. elliptic curves), not only RSA.

Reply:
Make sense however I am not familiar with different algorithms and I am afraid won't be able to implement the universal support here